### PR TITLE
Version 2

### PR DESCRIPTION
**Pull request by John-Atha**

### CODE REVIEW

File: setup/publish.sh [hunk 1]
```diff
@@ -35,7 +32,7 @@ switch_res=$(sui client switch --address ${ADMIN_ADDRESS})
 
 switch_env=$(sui client switch --env $1)
 
-publish_res=$(sui client publish --skip-fetch-latest-git-deps --gas-budget 2000000000 --json ${MOVE_PACKAGE_PATH})
+publish_res=$(sui client publish --skip-fetch-latest-git-deps --skip-dependency-verification --gas-budget 2000000000 --json ${MOVE_PACKAGE_PATH})
```
Comment (John-Atha) [inline setup/publish.sh:35]: The `skip-dependency-verification` flag is used temporarily

**Review by Eis-D-Z — APPROVED** [review]

File: move/blackjack/sources/single_player_blackjack.move [hunk 2]
```diff
@@ -318,26 +331,30 @@ module blackjack::single_player_blackjack {
...
+            id: object::new(ctx),
+            game_id: object::id(game),
+            current_player_sum: game.player_sum,
+        }
     }
 
     /// Function to be called by user who wants to stand.
     /// @param game: The Game object
-    public fun do_stand(game: &mut Game, player_hand_sum: u8, ctx: &mut TxContext) {
+    public fun do_stand(game: &mut Game, current_player_sum: u8, ctx: &mut TxContext): StandRequest {
```
Comment (Eis-D-Z) [inline move/blackjack/sources/single_player_blackjack.move:348]: Is the current_player_sum argument really necessary here? Is it used as some kind of double-check with the backend to make sure the data is in sync?
(same for above do_hit function)

**Review by Eis-D-Z — COMMENTED** [review]
On the player stake, it should be == to the STAKE. Since this is a demo it is a good practice in-line with the purpose of the app. If we want to refactor it for variable stake we can always do it later. We left it as fixed stake in Coinflip as well.

File: move/blackjack/sources/single_player_blackjack.move (hunk 2 above)
Comment (John-Atha) [reply move/blackjack/sources/single_player_blackjack.move:348]: yes yes that's the reason, as a simple double-check

File: SUI_DOCS.mdx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+For more details on building a back-end and deploying to Sui, check out the [Coin Flip app example](./coin-flip.mdx).
+
+You can also find the [full repository for this example here](https://github.com/MystenLabs/blackjack-sui).
+
+:::
+
+## Gameplay
+
+In this single-player version of blackjack, the player competes against a dealer, which is automated by the system. The dealer is equipped with a public BLS key that plays a central role in the game's mechanics. The dealer's actions are triggered by HTTP requests to serverless functions. Players generate randomness for the game by interacting with their mouse on the screen, after which they place their bet to start the game. Upon initiating the game, a request is made to the dealer's back-end, which processes it by signing and subsequently dealing two cards to the player and one to itself.
+
```
[the remark this answers is not in the document]
Comment (John-Atha) [reply SUI_DOCS.mdx:20]: updated, thanks!

File: SUI_DOCS.mdx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+The player has the options to 'Hit' or 'Stand.' Selecting 'Stand' triggers the dealer to draw cards until the total reaches 17 or higher. After the dealer stops, typically at a sum of 17 or more, the smart contract steps in to compare the totals and declare the winner. On the other hand, choosing 'Hit' prompts the dealer to draw an additional card for the player.
```
[the remark this answers is not in the document]
Comment (John-Atha) [reply SUI_DOCS.mdx]: updated, thanks!

File: SUI_DOCS.mdx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+| HTTP Method | Path                  | Description                                                   | Request Body                                      |
+| ----------- | --------------------- | ------------------------------------------------------------- | ------------------------------------------------- |
+| GET         | /api/health           | A simple healthcheck endpoint to check the API availability   | txDigest                                          |
+| POST        | /api/games/{id}/deal  | Executes the initial deal transaction after the game creation | txDigest                                          |
+| POST        | /api/games/{id}/hit   | Executes a hit move                                           | txDigest, id of corresponding HitRequest object   |
+| POST        | /api/games/{id}/stand | Executes a stand move                                         | txDigest, id of corresponding StandRequest object |
+
+### Need of usage of waitForTransactionBlock
+
+An interesting aspect of developing a Dapp on Sui, that is coupled to using a full node with/without a load balancer, and requires attention by the developer, is the occurence of write-after-write cases.
```
[the remark this answers is not in the document]
Comment (John-Atha) [reply SUI_DOCS.mdx]: updated, thanks!

File: SUI_DOCS.mdx [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+For testing purposes, the module provides special functions like `get_house_admin_cap_for_testing`, `player_won_post_handling_for_test` and `house_won_post_handling_for_test`, ensuring that developers can thoroughly test the game mechanics and house data handling.
+
+### Counter module
+
+The next module, [`counter_nft.move`](https://github.com/MystenLabs/blackjack-sui/blob/main/move/blackjack/sources/counter_nft.move), introduces the Counter NFT, a key component in the game's mechanics. It serves as the [Verifiable Random Function (VRF)](../cryptography/ecvrf.mdx) input, ensuring uniqueness in each game. The Counter NFT's value increases after each use, maintaining its distinctiveness for every new game. For first-time players, the creation of a Counter NFT is mandatory. To enhance user experience, the user interface can automate this process by integrating the Counter NFT's creation with the game initiation in a single transaction block. This seamless integration simplifies the process for the player, allowing them to focus on the gameplay. This counter serves the same purpose as the one in the [Coin Flip example](./coin-flip.mdx).
```
[the remark this answers is not in the document]
Comment (John-Atha) [reply SUI_DOCS.mdx]: updated, thanks!

**Review by teohaik — APPROVED** [review]

**Pull request merged**